native-audio/app/src/main/jni/native-audio-jni.c

```diff
@@ -243,10 +339,13 @@ void Java_com_example_nativeaudio_NativeAudio_createBufferQueueAudioPlayer(JNIEn
     (void)result;
 
     // get the effect send interface
-    result = (*bqPlayerObject)->GetInterface(bqPlayerObject, SL_IID_EFFECTSEND,
-            &bqPlayerEffectSend);
-    assert(SL_RESULT_SUCCESS == result);
-    (void)result;
+    bqPlayerEffectSend = NULL;
+    if( 0 == bqPlayerSampleRate) {
+        result = (*bqPlayerObject)->GetInterface(bqPlayerObject, SL_IID_EFFECTSEND,
+                                                 &bqPlayerEffectSend);
+        assert(SL_RESULT_SUCCESS == result);
+        (void)result;
+    }
 
 #if 0   // mute/solo is not supported for sources that are known to be mono, as this is
```

Comment:
my interpretation of this commented code:  when someone wants to create a multi-channel player,  she/he could easily un-comments the source and uses it right away.